### ISSUE TITLE
memory cgroup warning when launching RStudio Server of Ubuntu 22.04

### ISSUE THREAD

**Issue by kevinushey** [rstudio/rstudio#11235]
I saw this warning on on startup, using RStudio Server (devmode) on Ubuntu 22.04:

```
2022-05-18T20:36:20.981723Z [rsession-kevin] WARNING No memory control group found in /proc/self/cgroup; LOGGED FROM: std::string rstudio::core::system::{anonymous}::getMemoryCgroup() src/cpp/core/system/LinuxResources.cpp:335
```

This is what I saw for the process control group:

```
> system("cat /proc/self/cgroup")
0::/user.slice/user-1000.slice/user@1000.service/app.slice/app-org.gnome.Terminal.slice/vte-spawn-c5403501-9ce1-4d26-b869-0e8a7b95b619.scope
```

Note that RStudio Server (in devmode) here was indeed launched from a GNOME terminal, so it's plausible this issue might only affect development servers (or servers launched from a terminal).

Marking as developer for now as this likely will not affect production builds.

**Comment by ronblum** [comment]
It's in the production builds, too.

**Comment by jmcphers** [comment]
I don't think we should remove the warning outright, but maybe the check to see if CGroups memory stats are available needs to change for the layout on the virtual file system in Ubuntu 22.

**Comment by mikebessuille** [comment]
Tim could you have a look?  Maria suggests we might want to lower the severity of that message from Warning to Info or Debug, if we think it's not severe and since there's a successful fallback.

**Comment by boxswapper** [comment]
Isn't this a cgroup v2 issue.  I know that /sys/fs/cgroup/memory/memory.limit_in_bytes is available in cgroup v1

**Comment by jeffvroom** [comment]
Yes, I think we need to change the code to add support for situations where cgroup v2 is enabled, but v1 is not. We should also more cleanly handle the situation when neither is enabled. Perhaps a clear warning and disable the features that use it?  I'm not sure if this changed in Ubuntu 22.04, but I believe the default for even newer linux distros is to have both enabled in the kernel, but there's a configuration option to enable them individually so maybe the warning could point users to the right options to change the config.

**Comment by ronblum** [comment]
Verified that in RStudio Server 2022.06.0-daily+498 on Ubuntu 22.04 for which cgroup v2 is enableed. 
```
root@vmware:/home/vmware# mount | grep cgroup
cgroup2 on /sys/fs/cgroup type cgroup2 (rw,nosuid,nodev,noexec,relatime,nsdelegate,memory_recursiveprot)
```

Also double-checked memory in the Environment Pane, and it's functioning.

I'm leaving this open because I'd like to make sure that there's impact when cgroup v1 is used.

**Comment by ronblum** [comment]
Verified that works as expected with cgroup v1 (though tested on Ubuntu 20.04).
